### PR TITLE
Fixed space dragon's gust ability hitting people through walls

### PR DESCRIPTION
**Pull request by XeonMations**
## About The Pull Request

Fixed the space dragon's wind ability hitting people through walls. Also fixes a runtime oversight that made people not properly launched.

## Why It's Good For The Game

Being launched from behind a wall bad. 

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://github.com/user-attachments/assets/334a76c6-4941-4745-89bf-df04e8cd7258

</details>

## Changelog
:cl:XeonMations
balance: Made the space dragon's gust ability no longer hit people through walls.
fix: Also fixed a runtime happening with the space dragon's gust ability.
/:cl:

### CODE REVIEW

**Comment by Howluinb** [comment]
Space dragon is already weak enough, dragging a locker/hiding behind a wall to kite it infinitely is a massive nerf, why is this needed ?

**Comment by Rukofamicom** [comment]
I'm against this one. Space Dragon is already quite weak to ranged weaponry overall, and it's not implausible for the wind to go around corners in most cases due to the narrow space on stations.

**Comment by MarkusLarsson421** [comment]
Even if the dragon is weak against ranged weapons, being killed by "bullshit mechanics" will be annoying.

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
Cast the view from all adjacent tiles to the dragon. I don't want people to be able to move behind a single tile of cover while still being essentially next to the dragon to completely dodge the ability. The ability is relatively bad as-is due to the long charge period and your complete defenselessness while you charge.

File: code/modules/mob/living/simple_animal/hostile/space_dragon.dm [hunk 1]
```diff
@@ -414,15 +414,18 @@
...
+	for (var/turf/epicenter in view(1, usr.loc))
+		if(istype(epicenter, /turf/closed)) //Gusts dont go through walls.
 			continue
-		visible_message("<span class='boldwarning'>[candidate] is knocked back by the gust!</span>")
-		to_chat(candidate, "<span class='userdanger'>You're knocked back by the gust!</span>")
-		var/dir_to_target = get_dir(get_turf(src), get_turf(candidate))
-		var/throwtarget = get_edge_target_turf(target, dir_to_target)
-		candidate.safe_throw_at(throwtarget, 10, 1, src)
-		candidate.Paralyze(50)
+		for (var/mob/living/candidate in view(gust_distance, epicenter))
```
Comment (PowerfulBacon) [inline code/modules/mob/living/simple_animal/hostile/space_dragon.dm]: I'd add candidate to a list and check to prevent them being gusted 5 times in a row

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**